src/main/java/gregtech/common/metatileentities/multi/MetaTileEntityLargeBoiler.java

```diff
@@ -440,125 +190,85 @@ private boolean isFireboxPart(IMultiblockPart sourcePart) {
     @Override
     public ICubeRenderer getBaseTexture(IMultiblockPart sourcePart) {
         if (sourcePart != null && isFireboxPart(sourcePart)) {
-            return isActive ? boilerType.firefoxActiveRenderer : boilerType.fireboxIdleRenderer;
+            return isActive() ? boilerType.fireboxActiveRenderer : boilerType.fireboxIdleRenderer;
         }
-        return boilerType.solidCasingRenderer;
+        return boilerType.casingRenderer;
     }
 
     @Override
-    public boolean shouldRenderOverlay(IMultiblockPart sourcePart) {
-        return sourcePart == null || !isFireboxPart(sourcePart);
+    public boolean hasMufflerMechanics() {
+        return true;
     }
 
     @Override
-    public boolean onRightClick(EntityPlayer playerIn, EnumHand hand, EnumFacing facing, CuboidRayTraceResult hitResult) {
-        ItemStack itemStack = playerIn.getHeldItem(hand);
-        if (!itemStack.isEmpty() && itemStack.hasCapability(GregtechCapabilities.CAPABILITY_MALLET, null)) {
-            ISoftHammerItem softHammerItem = itemStack.getCapability(GregtechCapabilities.CAPABILITY_MALLET, null);
-
-            if (getWorld().isRemote) {
-                return true;
-            }
-            if (!softHammerItem.damageItem(DamageValues.DAMAGE_FOR_SOFT_HAMMER, false)) {
-                return false;
-            }
+    public void onAttached(Object... data) {
+        super.onAttached(data);
+        if (getWorld() != null && getWorld().isRemote) {
+            this.setupSound(GTSounds.BOILER, this.getPos());
         }
-        return super.onRightClick(playerIn, hand, facing, hitResult);
     }
 
-    public <T> T getCapability(Capability<T> capability, EnumFacing side) {
-        T result = super.getCapability(capability, side);
-        if (result != null)
-            return result;
-        if (capability == GregtechCapabilities.CAPABILITY_FUELABLE) {
-            return GregtechCapabilities.CAPABILITY_FUELABLE.cast(this);
+    @Override
+    protected void updateFormedValid() {
+        if (isMufflerFaceFree()) {
+            this.recipeLogic.update();
```

Comment:
Why is this needed, when the recipe logic update should already be ticking? Could this be what causes the double ticking issue that we were seeing and prevented using the `shouldUpdate` override?

Reply:
We need this to force the workable to not run if the muffler is blocked, otherwise muffler would be impactless. There are plenty of ways to do this, but this is the way we have been doing it in other places